src/app/static/src/app/components/downloadResults/UploadModal.vue

```diff
@@ -233,48 +274,67 @@
             ...mapStateProps<BaselineState, keyof Computed>("baseline", {
                 dataset: state => state.selectedDataset?.title
             }),
-            uploadFiles: mapStateProp<ADRUploadState, Dict<UploadFile>>("adrUpload",
+            createRelease(){
+                return this.choiceUpload === 'createRelease';
+            },
+            currentLanguage: mapStateProp<RootState, Language>(
+                null,
+                (state: RootState) => state.language
+            ),
+            uploadableFiles: mapStateProp<ADRUploadState, Dict<UploadFile>>("adrUpload",
                 (state) => state.uploadFiles!
             ),
             uploadFileSections() {
-                if (this.uploadFiles) {
-                    return Object.keys(this.uploadFiles).reduce((sections, key) => {
+                if (this.uploadableFiles) {
+                    return Object.keys(this.uploadableFiles).reduce((sections, key) => {
                         if (outputFileTypes.includes(key)) {
-                            sections[0][key] = this.uploadFiles[key];
+                            sections[0][key] = this.uploadableFiles[key];
                         } else {
-                            sections[1][key] = this.uploadFiles[key];
+                            sections[1][key] = this.uploadableFiles[key];
                         }
                         return sections;
                     }, [{} as any, {} as any]);
                 } else {
                     return [];
                 }
             },
+            uploadDisabled(){
+                return (!this.createRelease && this.uploadFilesToAdr.length < 1) || this.downloadingFiles;
```

Comment:
I wonder whether we should be disabling the upload button if you've selected the "create a release" option, but no files have changed since your last upload. Because in this case it will harmlessly but pointlessly go through the process of "uploading" all the files, but then, more problematically, fail to create a release. We know in advance that this is likely to fail, so allowing the user to do it is suboptimal. But then again I guess the dataset may have been changed by someone else in the meantime - so not quite so clear cut.

Reply:
I think in that case, where a release already exists for the latest point in the activity stream, we should at least make sure we show a comprehensible error message. I agree that the case where upload might succeed makes it not clear that we should disable upload - it's rather unlikely but could happen.

Reply:
The error message the user would get back in this scenario is quite helpful though. A disabled ok button with no other context might leave the user scratching their head and reporting it as a bug.

Reply:
Happy to leave as-is. Just wanted to raise as a potential optimisation - e.g. we could do `getReleases()` when the user starts an upload to potentially avoid pointlessly uploading all the files if nothing has changed since last release. But agree not for this ticket.